projects/WTI-UI/src/app/modules/core/services/contest.service.ts

```diff
@@ -74,5 +117,100 @@ export class ContestService extends IContestService {
+						}
+						//install the received contest clock data into the ContestService's ContestClock
+						this.installNewContestClock(data);					
+					}
+      			}, 
+				(error: any) => {
+        			console.error("ContestService.updateLocalContestClockFromServer(): getContestClock() subscription callback error: ");
+					console.error (error);
+      			}
+			);	
+	}
+	
+	/** This method receives a WTI-UI ContestClock model containing new values which should be used to update the WTI-UI contest clock,
+	 *  including the onscreen displays.  It constructs a new ContestClock object containing the received data and installs that
+	 *  object as the current WTI-UI clock.  It then updates the separate "ContestTimer" object with the specified values, and
+	 *  if the received data indicates the clock should be running it starts the ContestTimer (which then genrates a "clock tick"
+	 *  once per second to update the clock displays).
+	 */
+	installNewContestClock(data: any) {
+		
+		//copy the data fields (received from the PC2 Server via the WTI-API) into a new ContestService ContestClock object
+		let newContestClock = new ContestClock();
+		newContestClock.isRunning = data.running ;
+		newContestClock.contestLengthSecs = data.contestLengthInSecs ;
+		newContestClock.elapsedSecs = data.elapsedSecs ;
+		newContestClock.wallClockStartTime = data.wallClockStartTime ;
 
+		//save the new clock
+		if (DEBUG_MODE) {
+			console.log("ContestService (id", this.uniqueId, ").installNewContestClock(): replacing Contest Clock with:");
+			console.log(newContestClock);
+		}
+		this.contestClock = newContestClock;
+		
+		//pull the values out of the updated clock
+		let timerShouldBeStarted = this.contestClock.isRunning ;
+		let elapsedSecs = parseInt(this.contestClock.elapsedSecs);
+		let contestLengthSecs = parseInt(this.contestClock.contestLengthSecs);
+		let remainingSecs = contestLengthSecs - elapsedSecs;
+	
+		//shut off timer if it is running (otherwise we can't update the elapsed/remaining time values)
+		if (this.contestTimer.isTimerRunning) {
+			if (DEBUG_MODE) {
+				console.log("ContestService (id", this.uniqueId, ").installNewContestClock(): stopping timer");
+			}
+			this.contestTimer.stopTimer();
+		}
+		
+		//store the new contest time values in the Timer
+		this.contestTimer.setElapsedSecs(elapsedSecs);
+		this.contestTimer.setRemainingSecs(remainingSecs);
+		
+		//restart the timer if the new contest clock values indicate it should be running
+		if (timerShouldBeStarted) {
+			if (DEBUG_MODE) {
+				console.log("ContestService (id", this.uniqueId, ").installNewContestClock(): starting timer");
+			}
+			this.contestTimer.startTimer();
+		}
+	}
+	
+	/** Returns the number of seconds which have elapsed so far in the contest, which it obtains from the separate
+	 *  ContestTimer object.
+	 */
+	getElapsedSecs(): number {
+		let elapsedSecs = this.contestTimer.getElapsedSecs();	
```

Comment:
Is there a reason this is a "let", it can be const. Or it can be returned directly.

Reply:
Fixed in next push.